java/src/main/java/ai/rapids/cudf/ParquetColumnWriterOptions.java

```diff
@@ -0,0 +1,429 @@
+      return ParquetColumnWriterOptions.simpleColumnBuilder(name).build();
+    }
+
+    protected ParquetColumnWriterOptions withDecimal(String name, int precision) {
+      return ParquetColumnWriterOptions.simpleColumnBuilder(name)
+          .withDecimalPrecision(precision)
+          .build();
+    }
+
+    protected ParquetColumnWriterOptions withTimestamp(String name, boolean isInt96) {
+      return ParquetColumnWriterOptions.simpleColumnBuilder(name)
+          .withTimestampInt96(isInt96)
+          .build();
+    }
+  }
+
+  public static class StructBuilder extends NestedBuilder {
+
+    private List<ParquetColumnWriterOptions> children = new ArrayList<>();
+
+    /**
+     * Builder specific to build a Struct meta
+     * @param name
+     */
+    public StructBuilder(String name) {
+      this.name = name;
+    }
+
+    /**
+     * Set whether this struct column going to have nulls
+     * @param nullable
+     * @return this for chaining.
+     */
+    public StructBuilder withNullable(boolean nullable) {
+      this.nullable = nullable;
+      return this;
+    }
+
+    /**
+     * Set the list column meta.
+     * Lists should have only one child in ColumnVector, but the metadata expects a
+     * LIST column to have two children and the first child to be the
+     * {@link ParquetColumnWriterOptions#DUMMY_CHILD}.
+     * This is the current behavior in cudf and will change in future
+     * @param child
+     * @return this for chaining.
+     */
+    public StructBuilder withListChildColumn(ParquetListColumnWriterOptions child) {
+      assert (child.getChildColumnOptions().length == 2) : "Lists can only have two children";
+      if (child.getChildColumnOptions()[0] != DUMMY_CHILD) {
+        throw new IllegalArgumentException("First child in the list has to be DUMMY_CHILD");
+      }
+      if (child.getChildColumnOptions()[1].getColumName().isEmpty()) {
+        throw new IllegalArgumentException("Column name can't be empty");
+      }
+      children.add(child);
+      return this;
+    }
+
+    /**
+     * Set a child struct meta data
+     * @param child
+     * @return this for chaining.
+     */
+    public StructBuilder withStructChildColumn(ParquetStructColumnWriterOptions child) {
+      for (ParquetColumnWriterOptions opt: child.getChildColumnOptions()) {
+        if (opt.getColumName().isEmpty()) {
+          throw new IllegalArgumentException("Column name can't be empty");
+        }
+      }
+      children.add(child);
+      return this;
+    }
+
+    /**
+     * Set a simple child meta data
+     * @param name
+     * @return this for chaining.
+     */
+    public StructBuilder withSimpleChildColumn(String name) {
+      children.add(withColumnName(name));
+      return this;
+    }
+
+    /**
+     * Set a Decimal child meta data
+     * @param name
+     * @param precision
+     * @return this for chaining.
+     */
+    public StructBuilder withDecimalChildColumn(String name, int precision) {
```

Comment:
Recommend ditching the "Child" in the method names.  It's implied that we're adding a child column to anything we call `with...Column`, whether that's the top-level builder, a list builder, or a struct builder.